packages/ra-core/src/form/useInput.ts

```diff
@@ -153,7 +153,9 @@ export type InputProps<ValueType = any> = Omit<
 export type UseInputValue = {
     id: string;
     isRequired: boolean;
-    field: ControllerRenderProps;
+    field: Omit<ControllerRenderProps, 'onBlur'> & {
+        onBlur: (...event: any[]) => void;
```

Comment:
Why don't you use React's built-in FocusEventHandler?

Reply:
I relied on implementation of onBlur function in `useInput`
https://github.com/marmelab/react-admin/blob/e87f8f491b6e340084585fe0d6585aa8f710bc70/packages/ra-core/src/form/useInput.ts#L97-L102
and onBlur of InputProps where type is written in such way.

There are also 2 cases where the FocusEventHandler will fail the type contract.

1. A component needs to blur the input (BooleanInput, RichTextInput, FileInput, CheckboxGroupInput) because FocusEventHandler requires event as an argument.
```javascript
const handleChange = useCallback(
        event => {
            field.onChange(event);
            // Ensure field is considered as touched
            field.onBlur();
        },
        [field]
    );
```
2. RichTextInput provides EditorEvents['blur'] instead, which doesn't match with the FocusEvent.

Here is the solution I came up that matches type contracts. If you like it i'll commit it
```javascript
 field: Omit<ControllerRenderProps, 'onBlur'> & {
        onBlur: (
            event?:
                | FocusEvent<HTMLInputElement | HTMLTextAreaElement>
                | EditorEvents['blur']
        ) => void;
    };
```